### PR TITLE
Update/0.9.1

### LINKED ISSUE #237: 0.9.1 cairo support

### PR DESCRIPTION
**Pull request by war-in**
### **Please check if the PR fulfills these requirements**
- [x] The formatter, linter, and tests all run without an error
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

It adds changes from cairo 0.9.1

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

## Other information

(closes #237)

### CODE REVIEW

File: starknet_py/net/rpc_schemas/rpc_schemas.py [hunk 1]
```diff
@@ -170,6 +170,7 @@ class StarknetBlockSchema(Schema):
         data_key="transactions",
     )
     timestamp = fields.Integer(data_key="timestamp")
+    starknet_version = fields.String(data_key="starknet_version")
```
Comment (cptartur) [inline starknet_py/net/rpc_schemas/rpc_schemas.py:173]: I don't think the version of the RPC we support (0.15) has this fields. I think it'd be better to have it with `load_default` here.

File: starknet_py/net/client_models.py [hunk 2]
```diff
@@ -212,6 +234,7 @@ class BlockStateUpdate:
     old_root: int
     storage_diffs: List[StorageDiff]
     contract_diffs: List[ContractDiff]
+    declared_contracts: List[hex]
```
Comment (cptartur) [inline starknet_py/net/client_models.py:237]: I prefer we stick to using integers where possible

File: starknet_py/net/gateway_schemas/gateway_schemas.py [hunk 3]
```diff
@@ -206,10 +245,15 @@ def make_dataclass(self, data, **kwargs):
                 )
                 storage_diffs.append(storage_diff)
 
+        declared_contracts = data["state_diff"]["declared_contracts"]
```
Comment (cptartur) [inline starknet_py/net/gateway_schemas/gateway_schemas.py:248]: I'd prefer this had its' own schema like `ContractDiffSchema` and `StorageDiffSchema` even if it's just a single field.

**Review by Arcticae — CHANGES_REQUESTED** [review]
All seems good apart from that one

**Review by Arcticae — APPROVED** [review]

File: pyproject.toml [hunk 4]
```diff
@@ -15,14 +15,14 @@ documentation = "https://starknetpy.rtfd.io/"
...
 python = ">=3.7.2,<3.10"
-cairo-lang = "0.9.0"
 asgiref = "^3.4.1"
 sphinx = {version = "^4.3.1", optional = true}
 web3 = ">=5.28.0,<6"
 enum-tools = {extras = ["sphinx"], version = "0.9.0", optional = true}
 crypto-cpp-py = "^1.0.4"
 marshmallow = "^3.15.0"
 marshmallow-oneofschema = "^3.0.1"
+cairo-lang = "^0.9.1"
```
Comment (cptartur) [inline pyproject.toml:25]: I prefer we used specific cairo-lang version like it was done before.

**Review by cptartur — COMMENTED** [review]
I'm not quite sure about adding starknet_version to block as a optional field. Do you know if newer versions of rpc spec also added this field?

**Comment by ca11ab1e** [comment]
> I'm not quite sure about adding starknet_version to block as a optional field. Do you know if newer versions of rpc spec also added this field?

From the 0.9.1 patch, `starknet_version` was only added to `BlockInfo` (https://github.com/starkware-libs/cairo-lang/blob/13cef109cd811474de114925ee61fd5ac84a25eb/src/starkware/starknet/business_logic/state/state.py#L63).

**Comment by Pet3ris** [comment]
Yay

**Pull request merged**